com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/CreateAppStepTest.java

```diff
@@ -235,4 +242,140 @@ protected CreateAppStep createStep() {
         return new CreateAppStep();
     }
 
+    public static class TestWithDocker extends SyncFlowableStepTest<CreateAppStep> {
+
+        private ApplicationStagingUpdater applicationUpdater = Mockito.mock(ApplicationStagingUpdater.class);
+
+        private CloudApplicationExtended application;
+        private StepInput stepInput;
+        private DockerInfo dockerInfo;
+
+        public void setUp() {
+            loadParameters();
+            prepareContext();
+            prepareClient();
+        }
+
+        private void prepareContext() {
+            context.setVariable(Constants.PARAM_APP_ARCHIVE_ID, "archive_id");
+            context.setVariable(Constants.VAR_APPS_INDEX, stepInput.applicationIndex);
+            StepsUtil.setServicesToBind(context, mapToCloudServiceExtended());
+
+            byte[] serviceKeysToInjectByteArray = JsonUtil.toBinaryJson(new HashMap<>());
+            context.setVariable(Constants.VAR_SERVICE_KEYS_CREDENTIALS_TO_INJECT, serviceKeysToInjectByteArray);
+            stepInput.applications.get(0)
+                .setDockerInfo(dockerInfo);
+            StepsUtil.setAppsToDeploy(context, stepInput.applications);
+            StepsTestUtil.mockApplicationsToDeploy(stepInput.applications, context);
+        }
+
+        private List<CloudServiceExtended> mapToCloudServiceExtended() {
```

Comment:
Why is this method even necessary? When you're creating the application below, you're passing Collections.emptyList() as the list of services.